include/libdnf/advisory/advisory_query.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_ADVISORY_ADVISORY_QUERY_HPP
+#define LIBDNF_ADVISORY_ADVISORY_QUERY_HPP
+
+#include "libdnf/advisory/advisory.hpp"
+#include "libdnf/advisory/advisory_collection.hpp"
+#include "libdnf/advisory/advisory_package.hpp"
+#include "libdnf/advisory/advisory_reference.hpp"
+#include "libdnf/advisory/advisory_sack.hpp"
+#include "libdnf/common/sack/query_cmp.hpp"
+#include "libdnf/rpm/package.hpp"
+#include "libdnf/rpm/solv/solv_map.hpp"
+
+namespace libdnf::advisory {
+
+class AdvisoryQuery {
```

Comment:
May I ask you what do you plan about iterator?

Reply:
I wanted to discuss that, right now I am just getting a vector: https://github.com/rpm-software-management/libdnf/pull/1043/files#diff-286030ca3cb725841d1f8f86d804bdc3a70f6859895d65ddb001ffceac49136aR94-R96 to access the data. I am not sure whether we want to make AdvisorySet and its iterator (which AdvsoryQuery would inherit) or make specific iterator for AdvisoryQuery.

Reply:
AdvisorySet sound good to me. But in some cases we will use vector with advisories anyway.